pandas/core/algorithms.py

```diff
@@ -1776,6 +1820,43 @@ def take_nd(
 take_1d = take_nd
 
 
+def take_1d_array(
+    arr: np.ndarray,
+    indexer: np.ndarray,
+    out=None,
+    fill_value=lib.no_default,
+    allow_fill: bool = True,
+):
+    """
+    Specialized version for 1D arrays. Differences compared to take_nd/take_1d:
+
+    - Assumes input (arr, indexer) has already been converted to numpy arrays
+    - Only works for 1D arrays
+
+    """
+    if fill_value is lib.no_default:
+        fill_value = na_value_for_dtype(arr.dtype, compat=False)
+
+    if isinstance(arr, ABCExtensionArray):
```

Comment:
arr is annotated as ndarray

Reply:
Ah, good catch, the annotation is wrong (initially I was going to make a version that only dealt with ndarrays, because this check is expensive as well. But as long as the ArrayManager stores both, this check needs to be done somewhere). Also need to update the docstring for that.

Reply:
small perf boost if you check for `not isinstance(arr, np.ndarray)` i think

Reply:
Ah, yes, good idea. 
(being able to do ``isinstance(arr, ExtensionArray)`` would also already give a speedup compared to the ABC version (while still being more explicit), but that's difficult to import here I suppose)